miner/atx_pool.go

```diff
@@ -39,6 +40,9 @@ func (mem *AtxMemPool) GetAllItems() []*types.ActivationTx {
 func (mem *AtxMemPool) Put(atx *types.ActivationTx) {
 	mem.mu.Lock()
 	mem.atxMap[atx.Id()] = atx
+	if mem.listener != nil {
+		go mem.listener(atx.ActivationTxHeader)
```

Comment:
this is called for every atx we get in the pool, not only the one we are waiting for

Reply:
add a test for this case as well

Reply:
put is also called without any checks, so I think that potentially you can get `mem.listener` to be called twice and perhaps even panic with double close on the channel, WDYT?

Reply:
Yes. It's the listener's job to only act on the relevant ATX.

cc: @gavraz

Reply:
Only now noticed your last comment.

> put is also called without any checks

Not true, `Put` locks the mempool, see line 41.